File: YourChoice.Dominio/Entidades/ProdutoIngrediente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace YourChoice.Dominio.Entidades
{
    public class ProdutoIngrediente
    {
        public int ProdutoId { get; set; }
        public int IngredienteId { get; set; }
        public virtual Produto Produto { get; set; }
        public virtual Ingrediente Ingrediente { get; set; }
    }
}

File: YourChoice.Web/ClientApp/src/app/modelo/produtoIngrediente.ts
import { Ingrediente } from './ingrediente';
import { Produto } from './produto';
export class ProdutoIngrediente {
  produtoId: number;
  ingredienteId: number;
  produto: Produto[];
  ingrediente: Ingrediente[];

  constructor() {
    /*this.produtoIngredientes = [];*/
  }

}
